### ISSUE TITLE
Feature request: Add DNS-over-HTTP support to DNS Client

### ISSUE THREAD

**Issue by shawnkhall** [TechnitiumSoftware/DnsServer#518]
Currently when configuring DNS forwarding or rewrite rules the process is to configure the server using 127.0.0.1 then enable the `Optional Protocols` including `DNS-over-HTTP`/8053, then configure forwarding/rewrite to the 127.0.0.1 address. This can be a bit of a struggle since there's no included method to ensure that the DNS server itself is operating correctly on that IP and port combination.

For example, I had it rewrite working perfectly on IIS under Technitium DNS Server 7.x and 8.x, then had to completely change the setup for 9.x, and now it's failing again after upgrading to 10.x. Figuring out why it isn't working is a pain, to be polite.

Please add the DNS-over-HTTP/8053 option to the list at the DNS Client tab. This would enable immediate diagnostics to ensure that at least that component is working so I don't have to rely on another third party application to perform this testing.

**Comment by ShreyasZare** [comment]
Thanks for the post. You can already use the DNS Client with `http://127.0.0.1:8053/dns-query` as the server with protocol set to `DNS-over-HTTPS`. The protocol can use URL with both `http` and `https` scheme.

Regarding the issue you mentioned about reverse proxying to the `DNS-over-HTTP` service at 8053 port, there has been no change in the way it works. I have a similar setup with nginx that does the reverse proxying and never had to make any changes for any of these versions. Do let me know what changes you had to make in IIS so that I can understand the issue better.

**Comment by shawnkhall** [comment]
Thank you, @ShreyasZare , the `http://127.0.0.1:8053/dns-query` solution allowed me to troubleshoot the problem and resolve it. I simplified my rewrite pattern to use only `*` instead of regexp so hopefully this won't happen again. This might not have been necessary but hopefully it will improve performance.

In order to resolve the problem I modified the site binding in IIS once to break the site (I added 8053 which prevented it from loading in IIS) and again to reset them to what they should be (*:80 + *:443 using domain as host header) and it launched again and worked right away.

Perhaps all it really needed was the site being stopped and restarted in order to re-connect to the Technitium DNS Server on 8053 again, but `iisreset` did **not** fix it. 

Here's the rewrite rule I'm using:
![technitiumdnsserver-iis-rewrite-2022-12-08_22-33-11](https://user-images.githubusercontent.com/6037167/206641393-d2925fca-62ff-4d06-82a0-3788a4160a32.png)
